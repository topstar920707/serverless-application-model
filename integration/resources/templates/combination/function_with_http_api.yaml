Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: python3.7
      InlineCode: |
        def handler(event, context):
            return {'body': 'Hello World!', 'statusCode': 200}
      MemorySize: 128
      Events:
        GetApi:
          Type: HttpApi
          Properties:
            ApiId: 
              Ref: MyApi
            Method: GET
            Path: /some/path

  MyApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      DefinitionBody:
        info:
          version: '1.0'
          title:
            Ref: AWS::StackName
        paths:
          "/some/path": {}
          "/other": {}
        openapi: 3.0.1

Outputs:
  ApiUrl:
    Description: "API endpoint URL for Prod environment"
    Value:
      Fn::Sub: "https://${MyApi}.execute-api.${AWS::Region}.${AWS::URLSuffix}/"