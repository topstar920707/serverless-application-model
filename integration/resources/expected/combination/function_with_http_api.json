[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionGetApiPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyApi", "ResourceType":"AWS::ApiGatewayV2::Api" },
  { "LogicalResourceId":"MyApiApiGatewayDefaultStage", "ResourceType":"AWS::ApiGatewayV2::Stage" }
]